pkg/controllers/gameserverset/gameserverset_controller.go

```diff
@@ -225,7 +225,6 @@ func (r *GameServerSetReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return reconcile.Result{}, nil
 	}
 
-	// TODO sync PodProbeMarker
```

Comment:
In addition, this Reconcile method does not need List Pods, just Advanced StatefulSet Status can be implemented as well.

Reply:
When computing the number to scale,  Reconcile method need to list pods to get GameServer status info.